### PR TITLE
Handle reset signal from the admin app

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
Built on top of #187. Only the last commit matters.

### CODE REVIEW

File: src/card.rs [hunk 1]
```diff
@@ -51,6 +62,19 @@ impl<T: Client> Card<T> {
...
         reply: &mut heapless::Vec<u8, R>,
     ) -> Result<(), Status> {
+        if let Some(reset_signal) = self.state.volatile.reset_signal {
+            match reset_signal.load() {
+                ResetSignal::None => {}
+                ResetSignal::ConfigChanged => {
+                    return Err(Status::SelectedFileInTerminationState);
+                }
+                ResetSignal::FactoryReset => {
+                    self.state = State::default();
```
Comment (robin-nitrokey) [inline src/card.rs:72]: AFAIS this means we lose the reset signal.  Is this intended?

Comment (sosthene-nitrokey) [reply src/card.rs:72]: Good catch.
The volatile state is also lost on reset. I'll put the signal in the Options instead.

File: src/card.rs [hunk 2]
```diff
@@ -43,6 +44,16 @@ impl<T: Client> Card<T> {
         }
     }
 
+    /// Set the reset flag for the application
+    ///
+    /// If this flag is set it is loaded on every command, and a value of `true` will make opcard reject any
+    /// future command.
+    ///
+    /// It is meant as a way to "disable" the application until a reboot, for example to  change its backends
```
[the remark this answers is not in the document]
Comment (sosthene-nitrokey) [reply src/card.rs]: Removed

File: src/state.rs [hunk 3]
```diff
@@ -161,7 +161,7 @@ pub enum LifeCycle {
     Operational = 0x05,
 }
 
-#[derive(Clone, Debug, Default, Eq, PartialEq)]
+#[derive(Clone, Debug, Default)]
```
Comment (robin-nitrokey) [inline src/state.rs:164]: nit: no more need to remove the derives, but doesn’t make a difference

**Review by robin-nitrokey — COMMENTED** [review]
Please add a changelog entry and check if we need to clear the volatile state before resetting the state.  Otherwise LGTM!

**Review by robin-nitrokey — APPROVED** [review]
Thanks for the changes, looks good!

**Pull request merged**